app/modules/common/mocks/users.db.ts

```diff
@@ -47,43 +48,57 @@ export const users: User[] = [
     }
 ];
 
+
 @injectable()
 export class UsersDB {
 
-    constructor() {
-        console.log('UsersDB:constructor');
+    findAll(): User[] {
+        return cloneDeep(users);
     }
 
     findByLogin(login: string): User | undefined {
         if (!login) {
             return void 0;
         }
-        return users.find(item => item.login === login);
+        const user = users.find(item => item.login === login);
+        if (user) {
+            return cloneDeep(user);
+        }
+        return;
     }
 
     findById(id: number): User | undefined {
         if (!id) {
             return void 0;
         }
-        return users.find(item => item.id === id);
+        const user = users.find(item => item.id === id);
+        return cloneDeep(user);
+    }
+
+    findIndexById(id: number): number {
+        return users.findIndex(item => item.id === id);
     }
 
     create(user: User): User | undefined {
         const userEsists = this.findByLogin(user.login);
         if (userEsists) {
             return undefined;
         }
-        user.id = this.nextId;
-        users.push(user);
-        return user;
+        const userToSave = cloneDeep(user);
+        userToSave.id = this.nextId;
+        users.push(userToSave);
+        return userToSave;
     }
 
     update(id: number, data: User): User | undefined {
         const user = this.findById(id);
         if (!user) {
             return void 0;
         }
-        return Object.assign(user, data);
+        const updatedUser = Object.assign(user, data);
+        const index = this.findIndexById(user.id!);
+        users.splice(index, 1, updatedUser);
```

Comment:
`Object.assign()` mutates the first argument object, so you don't need to `splice`

Reply:
Now I see that `findById` returns the cloned user